File: chess-app/src/engine/pieces/Rook.js
import Piece from './Piece.js';

export default class Rook extends Piece {
    constructor(color) {
        super('rook', color);
    }

    getPseudoValidMoves(board, fromPos){
        let directions = [[1,0],[-1,0],[0,1],[0,-1]]
        return this._getSlidingMoves(directions, board, fromPos)
    }
}

File: chess/src/main/java/com/app/chess/Pieces/Rook.java
package com.app.chess.Pieces;
import java.util.List;

import com.app.chess.*;

public class Rook extends Piece {
    public Rook(Player color){
        super(PieceType.ROOK, color);
    }
    public Rook(){
        super(PieceType.ROOK);
    }

    @Override
    public List<int []> getPsuedoLegalMoves(Board board, int[] fromPos) {
        List<int[]> directions = List.of(
            new int[]{1,0}, new int[]{-1,0},
            new int[]{0,1}, new int[]{0,-1}
        );
        return getSlidingMoves(directions, board, fromPos);
    }

    @Override
    public Piece deepCopy() {
        Bishop copy = new Bishop(this.color);
        copy.hasMoved = this.hasMoved;
        return copy;
    }
}
